### PR TITLE
implement useQuery custom react hook

### LINKED ISSUE #40: Port over `useQuery` custom hook
`useQuery` from client project. 

Basically takes a record type and a params interface type as generic type arguments, re-runs the query and returns results whenever the query changes.

### PR DESCRIPTION
**Pull request by mrjones2014**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)
Fixes #40

### CODE REVIEW

File: src/interfaces/use-query-options.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { CatchResultHandler } from "andculturecode-javascript-core";
+import { ListServiceHook } from "../types/list-service-hook-type";
+import { NestedListServiceHook } from "../types/nested-list-service-hook-type";
+
+export interface UseQueryOptions<
+    TRecord,
+    TQueryParams,
+    TPathParams = undefined
+> {
+    service:
```
Comment (wintondeshong) [inline src/interfaces/use-query-options.ts:10]: Nitpick: Alphabetize and per commentary above, rename to better convey that it is a 'hook' by way of its name

**Review by wintondeshong — CHANGES_REQUESTED** [review]
@mrjones2014 thanks for another contribution! 👏 - if you wouldn't mind the version upgrade as well as rename of service, then I'm good 👍

File: src/hooks/use-query.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        service,
+        initialQuery,
+        initialPathParams,
+        onSuccess,
+        onError,
+    } = options;
+
+    const { list: listApi } = service();
+
+    const handleSuccess = useCallback(
```
[the remark this answers is not in the document]
Comment (mrjones2014) [reply src/hooks/use-query.ts:24]: They're using handlers passed in via options, that's why they need to be inside the hook.

**Comment by mrjones2014** [comment]
@wintondeshong comments should be addressed.

**Review by wintondeshong — APPROVED** [review]
Thanks for making those changes @mrjones2014

**Review by brandongregoryscott — APPROVED** [review]

**Comment by brandongregoryscott** [comment]
@all-contributors add @mrjones2014 for code

**Pull request merged**